index.js

```diff
@@ -7,256 +7,210 @@
- * //=> 'foo\.js'
- * ```
- * @param  {String} `str` String with `.`s to escape
- * @return {String}
- * @api public
- */
-
-config.escape = function(str) {
-  return str.replace(/\./g, '\\.');
-};
-
-/**
- * Return prop as a string. If an array is passed, a dot-notated
- * string will be returned.
- *
- * ```js
- * config.getPropString(['a', 'b']);
- * //=> 'a.b'
- * ```
- *
- * @param  {String|Array} `prop`
- * @return {String}
- * @api public
- */
-
-config.getPropString = function(prop) {
-  return Array.isArray(prop) ? prop.map(config.escape).join('.') : prop;
-};
-
-/**
- * Get a raw value from the project's Grunt configuration,
- * without processing `<% %>` template strings.
- *
- * ```js
- * config.getRaw([prop]);
- * ```
- * @param  {String} `prop` The name of the property to get.
- * @return {*} Returns the value of the given property.
- * @api public
- */
-
-config.getRaw = function(prop) {
-  if (prop) {
-    // Prop was passed, get that specific property's value.
-    return grunt.util.namespace.get(config.data, config.getPropString(prop));
-  } else {
-    // No prop was passed, return the entire config.data object.
-    return config.data;
-  }
-};
-
-/**
- * Match `<%= FOO %>` where FOO is a propString,
- * eg. `foo` or `foo.bar` but not a method call like
- * `foo()` or `foo.bar()`.
- */
-
-var propStringTmplRe = /^<%=\s*([a-z0-9_$]+(?:\.[a-z0-9_$]+)*)\s*%>$/i;
-
-/**
- * Get a value from the project's Grunt configuration, recursively
- * processing templates.
- *
- * ```js
- * config.set('a', 'b');
- * var foo = config.get('a');
- * //=> 'b'
- * ```
- * @param  {String} `prop`
- * @return {*} Returns the value of `prop`
- * @api public
- */
-
-config.get = function(prop) {
-  return config.process(config.getRaw(prop));
-};
-
-/**
- * Expand a config value recursively. Used for post-processing
- * raw values already retrieved from the config.
- *
- * ```js
- * config.set('a', 'b');
- * config.set('x', 'z');
- *
- * var foo = config.process('<%= a %>');
- * //=> 'b'
- * var bar = config.process(['<%= a %>', '<%= x %>']);
- * //=> ['a', 'z']
- * ```
- * @param  {String} `str`
- * @return {*} Resolved config values.
- * @api public
- */
-
-config.process = function(raw) {
-  return grunt.util.recurse(raw, function(value) {
-    // If the value is not a string, return it.
-    if (typeof value !== 'string') { return value; }
-    // If possible, access the specified property via config.get, in case it
-    // doesn't refer to a string, but instead refers to an object or array.
-    var matches = value.match(propStringTmplRe);
-    var result;
-    if (matches) {
-      result = config.get(matches[1]);
-      // If the result retrieved from the config data wasn't null or undefined,
-      // return it.
-      if (result != null) { return result; }
+var legacyUtil = require('grunt-legacy-util');
+var _ = require('lodash');
+
+exports.create = function(options) {
+  options = _.extend({}, {grunt: null, log: null}, options);
```

Comment:
Just expect a `grunt` option, `log.wordlist`, `log.error` `verbose` and `template.process` will be pulled from it.